### PR TITLE
feat: add ES2018 migration blog post

### PR DESCRIPTION
**Pull request by barmac**
This PR adds a blog post on the ES2018 syntax migration in our core libraries.

### CODE REVIEW

File: src/blog/2022/009-es2018.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+export default {
+  input: 'src/main.js',
+  output: {
+    file: 'bundle.js',
+    format: 'cjs'
+  },
+  plugins: [resolve(), babel({ babelHelpers: 'bundled' })]
+};
+```
```
[the remark this answers is not in the document]
Comment (barmac) [reply src/blog/2022/009-es2018.md:45]: Hmm I could extend the bundling example

[the remark this answers is not in the document]
Comment (barmac) [reply src/blog/2022/009-es2018.md:45]: I've added a link to the example PR for webpack config.

File: src/blog/2022/009-es2018.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+---
+
+title: Core Libraries Migrated To ES2018
```
Comment (nikku) [inline src/blog/2022/009-es2018.md:3]: I'd simplify this to `Migration to ES2018`. Not sure if it is clear what the "core libraries" are.

File: src/blog/2022/009-es2018.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+---
+
+<p class="introduction">
+  We are happy to announce the latest releases of our modeling toolkit which ship with modern JavaScript syntax, namely ES2018.
+</p>
+
+<!-- continue -->
+
+The releases introduce modern language constructs to the codebase, i.e. async/await, ES6-style classes, as well as rest/spread operator. We believe this will allow reduce the burden on contributors who won't need to limit to ES5 syntax anymore.
```
Comment (nikku) [inline src/blog/2022/009-es2018.md:24]: I'd want us to clarify here that the _bundled_ output will contain ES2018 features. You could also say that the target runtime of our libraries is now ES2018 (which is every modern browser).

The main benefit for the user is smaller bundles (we don't shim/transpile back to ES5), while the main benefit on our end is maintainability and the ability to use these new language features freely.

File: src/blog/2022/009-es2018.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+## Migration
+
+If your application does not target unsupported browsers (mainly Internet Explorer 11), you don't need to do anything to prepare for the new versions. Also, if you already transpile your code, no additional steps should be required.
```
Comment (nikku) [inline src/blog/2022/009-es2018.md:28]:
```suggestion
Unless you need to support ancient browsers such as Internet Explorer 11, you don't need to do anything to prepare for the new versions. Also, if you already transpile your code using tools such as [Babel](https://babeljs.io/), no additional steps are required.
```

File: src/blog/2022/009-es2018.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Cf. [rollup guide](https://rollupjs.org/guide/en/#babel).
+
+If your bundler of choice is [webpack](https://webpack.js.org/), check out [the bundling example](https://github.com/bpmn-io/bpmn-js-examples/pull/184).
+
+## Wrapping up
+
+Did we miss anything? Did you spot a bug, or would you like to suggest an improvement?
+
+Reach out to us via [our forums](https://forum.bpmn.io/), tweet us [@bpmn_io](https://twitter.com/bpmn_io) or file an issue you found in the respective issue tracker:
```
Comment (nikku) [inline src/blog/2022/009-es2018.md:55]:
```suggestion
Reach out to us via [our forums](https://forum.bpmn.io/), tweet us [@bpmn_io](https://twitter.com/bpmn_io) or file an issue you found in the respective issue trackers for [bpmn-js](...), [dmn-js](...), or [diagram-js](...).
```

**Comment by nikku** [comment]
I overall like this post.

What I'd love us to convey: We're migrating the whole eco-system to ES2018 as a library _target_ (what does this mean? => you have to transpile + shim what we publish if you'd like it to work in ES5 environments (ancient browsers)). We make the first big move by migrating some of our important toolkits (to ensure this change is recognized).

File: src/blog/2022/009-es2018.md (hunk 3 above)
Comment (barmac) [reply src/blog/2022/009-es2018.md]: Please check now.

**Comment by nikku** [comment]
Much better now. I tried to make the post just a little more lively via https://github.com/bpmn-io/bpmn.io/pull/147/commits/b05d9555dae45631efdcf125f1b0b46812941fd8. Please have a look. Could not hold back from the IE is DEAD thing.

Consider merging https://github.com/bpmn-io/bpmn.io/pull/147/commits/b05d9555dae45631efdcf125f1b0b46812941fd8 or re-mixing what you see fit.

**Comment by barmac** [comment]
Thanks for the review. I am merging now the blog post and will publish it to live as soon as we are ready to release the libs.

**Pull request merged**